src/figuresClasses.ts

```diff
@@ -1,19 +1,113 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape : 'triangle' | 'circle' | 'rectangle';
+  color : Color;
+  getArea() : number;
 }
 
-export class Triangle {
+let square : number;
 
-}
+export class Triangle implements Figure {
+  shape : 'triangle';
+
+  color : 'red' | 'green' | 'blue';
+
+  a: number;
+
+  b: number;
+
+  c: number;
 
-export class Circle {
+  constructor(color : 'red' | 'green' | 'blue',
+    a : number, b : number, c : number) {
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
use this https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/